plugin/acl/README.md

```diff
@@ -0,0 +1,98 @@
+# acl
+
+*acl* - enforces basic access control policies and prevents unauthorized access to protected servers.
+
+## Description
+
+With `acl` enabled, users are able to filter DNS queries with filtering rules, i.e. allowing authorized queries to recurse or blocking unauthorized queries towards protected DNS zones.
+
+This plugin can be used multiple times per Server Block.
+
+## Syntax
+
+```
+acl [ZONES…] {
+    ACTION type QTYPE net SOURCE
+    ACTION type QTYPE file LOCAL_FILE
+}
+```
+
+- **ZONES** zones it should be authoritative for. If empty, the zones from the configuration block are used.
+- **ACTION** (*allow* or *block*) defines the way to deal with DNS queries matched by this rule. The default action is *allow*, which means a DNS query not matched by any rules will be allowed to recurse.
+- **QTYPE** is the query type to match for the requests to be allowed or blocked. Common resource record types are supported. *ANY* stands for all kinds of DNS queries.
+- **SOURCE** is the source ip to match for the requests to be allowed or blocked. Typical CIDR notation and single IP address are supported. *ANY* stands for all possible source IP addresses.
+- **LOCAL_FILE** is a local file including a list of IP addresses or subnets. It allows users to import blacklist/whitelist from third party directly without setting rules by hand.
```

Comment:
I think the local file functionality can be done with standard Corefile constructs like import - if so this entire functionality can be removed.

Reply:
Agree that this functionality overlaps a little bit with plugin `import`. The reason of introducing `LOCAL_FILE` is support the common use of IP filter, to block a list of ip address.

Yes `import` can do the same thing but seems it requires some efforts in formatting? like putting all ip addresses in one line and stripping all comments.